lib/OpenQA/Utils.pm

```diff
@@ -682,5 +683,36 @@ sub mark_job_linked {
     }
 }
 
+sub detect_current_version($) {
+    my ($path) = @_;
+    # Get application version
+    my $current_version = undef;
+
+    if (-e $path . '/public/Changelog') {
+        my $changelog = file_content($path . '/public/Changelog');
+        if ($changelog && $changelog =~ /Update to version (\d+\.\d+\.\d+\.(\b[0-9a-f]{5,40}\b))\:/mi) {
+            $current_version = $1;
+        }
+    }
+    elsif (-e $path . '/.git/refs/heads/master' && -e $path . '/.git/packed-refs') {
+        my $master_head = file_content($path . '/.git/refs/heads/master');
```

Comment:
Tiny nitpick, `Mojo::File::slurp` is faster than `file_content`, so this would be a little more efficient.
```
use Mojo::File 'path';
...
my $master_head = path($path, '.git', 'refs', 'heads', 'master')->slurp;
```